### PR TITLE
AArch64: Implementation of awrtbariEvaluator in AArch64

### PR DESCRIPTION
**Pull request by mflawn**
Work in progress implementation of awrtbariEvaluator and associated helper.
wrtbarEvaluator will also be used by awrtbarEvaluator when that is implemented.

Signed-off-by: Michael Flawn <mflawn@unb.ca>

### CODE REVIEW

**Comment by mflawn** [comment]
This evaluator was built successfully for AArch64 and is largely adapted from power, I have a few associated questions. See my other WIP PR on awrtbarEvalutator for further possibly relevant questions.
Comments/Questions:
```
1) Since fieldwatch is unavailable for ARM do we need valueNode or anything associated with it?
2) dmb seems to be unimplemented (but dsb might be?) so should the needSync checks be left out and/or commented out? 
3) Do we need any kind of AArch64 equivalent to postSyncConditions?
4) cg->insertPrefetchIfNecessary does nothing by default, except in power, I'm guessing this should be removed unless this was something intended to be implemented later. 
   Unless otherwise specified I will remove this in the final product.
```

**Comment by mflawn** [comment]
I changed this to reflect Daryl's comments on awrtbarEvaluator, as some of the same things apply here. I also cleaned up tempMR and removed the sizeofMR variable.

File: runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -101,6 +101,33 @@ J9::ARM64::TreeEvaluator::generateFillInDataBlockSequenceForUnresolvedField(TR::
    TR_ASSERT_FATAL(false, "This helper implements platform specific code for Fieldwatch, which is currently not supported on ARM64 platforms.\n");
    }
 
+static void wrtbarEvaluator(TR::Node *node, TR::Register *srcReg, TR::Register *dstReg, bool srcNonNull, bool needDeps, TR::CodeGenerator *cg)
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp]: This implementation isn't needed in this PR as it was merged as part of #6495.  Please remove.  That is, assuming they are identical...

Comment (mflawn) [reply runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp]: Removed previously merged function wrtbarEvaluator.

**Comment by 0xdaryl** [comment]
This is an AArch64-specific change.  A Jenkins build is not required.

**Pull request merged**